assets/css/editor.scss

```diff
@@ -48,6 +48,15 @@
 	}
 }
 
+// Display white icon when block is selected in list view.
+.block-editor-list-view-leaf.is-selected {
+	.block-editor-list-view-block-contents {
+		svg {
+			fill: #fff;
+		}
+	}
+}
```

Comment:
Taking a closer look, I think we should do a couple of changes:

* Ideally, we should make sure these styles only apply to WooCommerce icons. I think the only way of doing so is adding a class to each `Icon` instance used as the block icon. So `<Icon icon={ box } color="#7f54b3" />` would become something like `<Icon className="wc-block-editor-components-block-icon" icon={ box } color="#7f54b3" />`. Then, we could use that class name in the CSS.
At this point, it might even make sense creating a `<BlockIcon />` component that encapsulates the class name and the `color` attribute so we don't need to repeat it for each block.
* We should use [`currentColor`](https://css-tricks.com/currentcolor/) instead of `#fff`, this is what Gutenberg is currently doing and it's more future-proof if they ever change the colors.

What do you think about these proposals @nielslange? I know they increase the scope of this PR so I'm ok with leaving it for a follow-up, but I think they would make our code more resilient for the future.

Reply:
> What do you think about these proposals @nielslange? I know they increase the scope of this PR so I'm ok with leaving it for a follow-up, but I think they would make our code more resilient for the future.

That's a great idea, @Aljullu! To keep this PR lean, I created https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/5355. I believe adding your suggestions to this PR will shift the scope away from the initial case. Nevertheless, it's a great idea. I'll merge this PR for now and will implement the following-up PR later today.